app/src/components/Context/index.tsx

```diff
@@ -2,19 +2,23 @@ import { AuthManager } from '@/lib/auth';
 import { EditorEngine } from '@/lib/editor/engine';
 import { ProjectsManager } from '@/lib/projects';
 import { RouteManager } from '@/lib/routes';
+import { UpdateManager } from '@/lib/update';
 import { createContext, useContext } from 'react';
 
 const authManager = new AuthManager();
 const routeManager = new RouteManager();
 const projectsManager = new ProjectsManager();
 const editorEngine = new EditorEngine(projectsManager);
+const updateManager = new UpdateManager();
 
 const AuthContext = createContext(authManager);
 const RouteContext = createContext(routeManager);
 const ProjectsContext = createContext(projectsManager);
 const EditorEngineContext = createContext(editorEngine);
+const UpdateContext = createContext(updateManager);
 
 export const useAuthManager = () => useContext(AuthContext);
 export const useRouteManager = () => useContext(RouteContext);
 export const useProjectsManager = () => useContext(ProjectsContext);
 export const useEditorEngine = () => useContext(EditorEngineContext);
+export const useUpdateManager = () => useContext(UpdateContext);
```

Comment:
Wtf, was this always missing?

Reply:
Seems like it! I tossed it in because I couldn't get the update context without it.

Reply:
Ah I see...

Reply:
It is duplicated. I like your pattern better so I'll remove the existing one.